packages/sanity/src/core/store/events/getDocumentTransactions.ts

```diff
@@ -0,0 +1,91 @@
+import {type TransactionLogEventWithEffects} from '@sanity/types'
+import {type SanityClient} from 'sanity'
+
+import {getJsonStream} from '../_legacy/history/history/getJsonStream'
+
+const TRANSLOG_ENTRY_LIMIT = 50
+
+const documentTransactionsCache: Record<string, TransactionLogEventWithEffects[]> =
+  Object.create(null)
+
+// Transactions could be cached, given they are not gonna change EVER.
```

Comment:
Yes, what makes it a bit challenging and why I created another ticket to do this change is that we could reuse transactions that we have already fetched.
Given transactions are not changing and we know the order.

For example:

1) User inspect the diff from `rev1` to `rev10` , we load all the transactions between this two revisions.

2) User inspect the diff from `rev4` to `rev5`, we are loading again the transactions between this two, but we could reuse the ones loaded in the first diff, given they are contained by this.

3) User inspects `rev8` to `rev15` , we are loading from rev8, we could avoid this and load only from rev10 because we already have from 8 to 10 loaded from the action 1